test/connectors/CompanyHouseConnectorSpec.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import base.SpecBase
+import models.{Submission, SubmissionResponse}
+import org.scalatest.concurrent.ScalaFutures
+import play.api.libs.json.{JsValue, Json}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, UpstreamErrorResponse}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import utils.MockUserAnswers
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import org.mockito.Mockito._
+import org.mockito.ArgumentMatchers.{any, eq => eqTo}
+
+class CompanyHouseConnectorSpec extends SpecBase with ScalaFutures {
+
+  "submission" must {
+
+    "return true when the HTTP call to CoHo is a match" in {
+
+      implicit val hc: HeaderCarrier = HeaderCarrier()
+
+      val answers = MockUserAnswers.minimalValidUserAnswers
+
+
+      val httpMock = mock(classOf[HttpClient])
+      when(httpMock.GET[HttpResponse](any())(any(), any(), any()))
+        .thenReturn(Future.successful(HttpResponse(200, Some(Json.parse("""{"links":{"filing_history":"/company/15428444/filing-history","officers":"/company/15428444/officers","persons_with_significant_control_statement":"/company/15428444/persons-with-significant-control-statement","self":"/company/15428444"},"accounts":{"next_due":"2022-12-24","next_accounts":{"period_start_on":"2021-03-24","period_end_on":"2022-03-24","due_on":"2022-12-24","overdue":false},"next_made_up_to":"2022-03-24","accounting_reference_date":{"day":"24","month":"3"}},"company_number":"15428444","date_of_creation":"2021-03-24","type":"ltd","undeliverable_registered_office_address":false,"company_name":"Company 15428444 LIMITED","sic_codes":["71200"],"confirmation_statement":{"next_made_up_to":"2022-03-24","overdue":false,"next_due":"2022-04-07"},"registered_office_is_in_dispute":false,"company_status":"active","etag":"857b47e351ef2dad63f3734d6eac3440aa5aab28","has_insolvency_history":false,"registered_office_address":{"address_line_1":"Companies House","address_line_2":"Crownway","country":"United Kingdom","locality":"Cardiff","postal_code":"CF14 3UZ"},"jurisdiction":"england-wales","has_charges":false,"can_file":true}""")))))
```

Comment:
This is using a deprecated apply method for HttpResponse, you can remove the ```Some(Json.parse(``` (and 2 parentheses at the end of the line) to just pass it in as a string and avoid that (same for other tests in the file)